a3p-integration/proposals/z:acceptance/test-lib/psm-lib.js

```diff
@@ -0,0 +1,492 @@
+  }
+
+  const offerSpec = {
+    id: offerId,
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer: charterAcceptOfferId,
+      invitationMakerName: 'VoteOnParamChange',
+    },
+    proposal: {},
+    offerArgs: {
+      instance: instances[instanceName],
+      params,
+      deadline: BigInt(deadline * 60 + Math.round(Date.now() / 1000)),
+    },
+  };
+
+  /** @type {string | object} */
+  const spendAction = {
+    method: 'executeOffer',
+    offer: offerSpec,
+  };
+
+  const offer = JSON.stringify(marshaller.toCapData(harden(spendAction)));
+  console.log(offerSpec);
+  console.log(offer);
+
+  return executeOffer(address, offer);
+};
+
+/**
+ *
+ * @param {{
+ *   committeeAddrs: Array<string>
+ *   position: number | string
+ * }} VotingDetails
+ */
+export const voteForNewParams = ({ committeeAddrs, position }) => {
+  console.log('ACTIONS voting for position', position, 'using', committeeAddrs);
+  return Promise.all(
+    committeeAddrs.map(account =>
+      // @ts-expect-error Casting
+      agops.ec('vote', '--forPosition', position, '--send-from', account),
+    ),
+  );
+};
+
+/**
+ * @typedef {{
+ *   giveMintedFeeVal: bigint;
+ *   wantMintedFeeVal: bigint;
+ *   mintLimit: bigint;
+ * }} Params
+ *
+ *
+ * @param {{
+ *   address: string
+ *   instanceName: string
+ *   newParams: Params
+ *   deadline: number
+ *   offerId?: string
+ * }} question
+ *
+ * @param {{
+ *   committeeAddrs: Array<string>
+ *   position: number
+ * }} voting
+ */
+export const implementPsmGovParamChange = async (question, voting) => {
+  await buildProposePSMParamChangeOffer(question);
+  await voteForNewParams(voting);
+  console.log('ACTIONS wait for the vote deadline to pass');
+  await new Promise(r => setTimeout(r, VOTING_WAIT_MS));
+};
+
+export const checkGovParams = async (t, expected, psmName) => {
+  const current = await getPsmGovernance(psmName);
+
+  t.log({
+    give: current.WantMintedFee.value,
+    want: current.GiveMintedFee.value,
+    mintLimit: current.MintLimit,
+  });
+
+  t.like(current, expected);
+};
+
+/**
+ *
+ * @param {string} name
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} fund
+ */
+export const initializeNewUser = async (name, fund) => {
+  const psmTrader = await addUser(name);
+  await Promise.all([
+    bankSend(psmTrader, `20000000ubld,${fund.value}${fund.denom}`),
+    bankSend(psmTrader, `1000000uist`, GOV1ADDR),
+  ]);
+};
+
+/**
+ *
+ * @param {any} t
+ * @param {string} userName
+ * @param {{
+ *   denom: string,
+ *   value: string
+ * }} expectedAnchorFunds
+ */
+export const checkUserInitializedSuccessfully = async (
+  t,
+  userName,
+  expectedAnchorFunds,
+) => {
+  let userAddress;
+  await t.notThrowsAsync(async () => {
+    userAddress = await getUser(userName);
+  });
+
+  // @ts-expect-error initialization
+  const balance = await getBalances([userAddress], expectedAnchorFunds.denom);
+  t.is(balance >= BigInt(expectedAnchorFunds.value), true);
```

Comment:
this also doesn't have a diagnostic message. I think a simple `assert` would be more appropriate here, so you don't have to thread the `t` (especially since the `any` erases the type)
```suggestion
  assert(balance >= BigInt(expectedAnchorFunds.value));
```

Reply:
resolved in https://github.com/Agoric/agoric-sdk/commit/712f97060737ce2a54d4e27ba2047142da6585ab